test/common/test_polygon_mesh.cpp

```diff
@@ -91,11 +91,12 @@ TEST(PolygonMesh, concatenate_cloud)
 
 TEST(PolygonMesh, concatenate_vertices)
 {
+    const std::size_t size = 15;
+
     PolygonMesh test, dummy;
-    test.cloud.width = 10;
-    test.cloud.height = 5;
+    test.cloud.width = dummy.cloud.width = size;
+    test.cloud.height = dummy.cloud.height = 1;
```

Comment:
Thank you for pointing out this point. In short,
1. The size of the dummy cloud here (which wasn't set before) must be set correctly;
2. No matter we set the height, the concatenated cloud (`test`) will have 1 as its height.

We see here `pcl::PCLPointCloud2::concatenate(...)` is called inside `pcl::PolygonMesh::concatenate(...)`:
https://github.com/PointCloudLibrary/pcl/blob/e1664a4ba63107b2c5ca10bc4137c5a8f73ae0b5/common/include/pcl/PolygonMesh.h#L30-L33

Both the sizes of the clouds are computed here (L58-L59):
https://github.com/PointCloudLibrary/pcl/blob/e1664a4ba63107b2c5ca10bc4137c5a8f73ae0b5/common/src/PCLPointCloud2.cpp#L48-L59

And they are used in calculating the new size. Note that the height is set 1.
https://github.com/PointCloudLibrary/pcl/blob/e1664a4ba63107b2c5ca10bc4137c5a8f73ae0b5/common/src/PCLPointCloud2.cpp#L142-L143

So, if we omit the initialization of the size of our dummy cloud, it will yield the wrong size of the concatenated cloud.
Plus, the height is just sufficient to be 1 than any other numbers.

Reply:
Thanks. Makes sense. Can we add a comment saying that the original cloud being organized or unorganized doesn't affect the algorithm?